api/maven-api-core/src/main/java/org/apache/maven/api/services/RuntimeInformation.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.api.services;
+
+import org.apache.maven.api.Service;
+import org.apache.maven.api.Version;
+import org.apache.maven.api.annotations.Experimental;
+import org.apache.maven.api.annotations.Nonnull;
+
+/**
+ * Service interface to provide runtime information.
+ *
+ * @since 4.0.0
+ */
+@Experimental
+public interface RuntimeInformation extends Service {
+
+    /**
+     * Returns the "runtime version", Maven version caller runs in.
+     *
+     * @return the runtime version, never {@code null}
+     */
+    @Nonnull
+    Version runtimeVersion();
```

Comment:
We already have `Session.getMavenVersion()`, I'm not sure we need a new service just for this bit.  However the session's method returns a `String`, maybe it needs to be changed to a `Version`.  If we keep this service, then the `Session.getMavenVersion` method should delegate to that Service.
I think I did not used a service, I did not see the value for that single method...

Reply:
Will drop this and make `Session.getMavenVersion()` return a `Version`, as the most cases RT version is needed is:
* display purposes (asString() is there)
* compare/range check, and Version is needed for that

Having `Session.getMavenVersion()` return plain String is ok for 1st but not enough for 2nd use case.